### DISCUSSION TITLE
Run pip install -e .. in path=model/model_training, mac m1 meet error

### DISCUSSION THREAD

**Discussion by OrangeJessie** [LAION-AI/Open-Assistant#3352] [Q&A]
Here is the issue, seems like GPU can't find. How can I solve this issue?
I read the historical issues for mac, but don't get anything helpful.

error: subprocess-exited-with-error

  × python setup.py egg_info did not run successfully.
  │ exit code: 1
  ╰─> [16 lines of output]
      Traceback (most recent call last):
        File "<string>", line 2, in <module>
        File "<pip-setuptools-caller>", line 34, in <module>
        File "/private/var/folders/1y/s1tgwg0d4rqc_nrfwm2f_r140000gp/T/pip-install-u3qof7be/flash-attn_7f22639ed7b743ae9d3555e587d38c77/setup.py", line 83, in <module>
          _, bare_metal_major, bare_metal_minor = get_cuda_bare_metal_version(CUDA_HOME)
        File "/private/var/folders/1y/s1tgwg0d4rqc_nrfwm2f_r140000gp/T/pip-install-u3qof7be/flash-attn_7f22639ed7b743ae9d3555e587d38c77/setup.py", line 23, in get_cuda_bare_metal_version
          raw_output = subprocess.check_output([cuda_dir + "/bin/nvcc", "-V"], universal_newlines=True)
      TypeError: unsupported operand type(s) for +: 'NoneType' and 'str'

      Warning: Torch did not find available GPUs on this system.
       If your intention is to cross-compile, this is not an error.
      By default, We cross-compile for Volta (compute capability 7.0), Turing (compute capability 7.5),
      and, if the CUDA version is >= 11.0, Ampere (compute capability 8.0).
      If you wish to cross-compile for a single specific architecture,
      export TORCH_CUDA_ARCH_LIST="compute capability" before running setup.py.

      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

**Comment by AbdBarho** [comment]
I don't think any of our ML code is intended to run on any device other than Nvidia GPUs.

**Comment by OrangeJessie** [comment]
> I don't think any of our ML code is intended to run on any device other than Nvidia GPUs.

IC, thank you

**Comment by a1nkit** [comment]
@OrangeJessie Getting similar errors on google colab.

I have set GPU in runtime (Nvidia Tesla T4). 

I believe as per your earlier comment, it should have worked on above google colab notebook .

I am adding the link to my notebook for reference: https://colab.research.google.com/drive/1guc1y2CBzA4BN6h6XHVZYLZlIr1JLSIx?usp=share_link

Hoping to get a reply.

Thank you.

**Reply by AhmedKhaled945** [reply]
any luck solving that? @a1nkit
